src/main/java/jenkins/plugins/http_request/HttpRequest.java

```diff
@@ -281,19 +327,25 @@ private HttpRequestBase getHttpRequestBase(PrintStream logger, RequestAction req
         return httpRequestBase;
     }
 
-    private FilePath getOutputFilePath(AbstractBuild<?, ?> build) {
+    private FilePath getOutputFilePath(FilePath workspace) {
         if (outputFile != null && !outputFile.isEmpty()) {
-            return build.getWorkspace().child(outputFile);
+            return workspace.child(outputFile);
         }
         return null;
     }
 
     private List<NameValuePair> createParameters(
-            AbstractBuild<?, ?> build, PrintStream logger,
+            Run<?, ?> run, PrintStream logger,
             EnvVars envVars) {
         if (!passBuildParameters) {
             return Collections.emptyList();
         }
+
+        if (!(run instanceof AbstractBuild<?, ?>)) {
+            return Collections.emptyList();
```

Comment:
This code adds all build parameters to the url. With Workflow, the user can cherry pick which parameters are added to the url as they construct the url themselves. You are right, I should document this.